pennylane/qnode.py

```diff
@@ -851,6 +851,14 @@ def __call__(self, *args, **kwargs):  # pylint: disable=too-many-branches, too-m
 
         # construct the tape
         self.construct(args, kwargs)
+        if override_shots is not False:
+            self._tape.shots = override_shots
+        else:
+            self._tape.shots = (
+                self.device._raw_shot_sequence
+                if self.device._has_partitioned_shots()
+                else self.device.shots
+            )

```

Comment:
I prefer to always specify the properties on initialization. This will set us up to make these objects immutable in the future. So instead we can modify `make_qscript` to be able to accept a shots argument, which is then used upon initialization of the `QuantumScript`.

Reply:
I've updated `make_qscript` to take a `shots` argument. I've also updated `QNode.__call__` and `QNode.construct`. I feel a bit weird about the changes to `QNode` because I put shots into the qfunc `kwargs` and then remove them before `make_qscript` is called. I did this to avoid changing the interface of the updated methods, but if people are fine with it I can just change the interface of `QNode.construct`.

Reply:
Changes to `QNode` have been reverted.